server/app/build.gradle

```diff
@@ -0,0 +1,131 @@
+apply plugin: 'com.android.application'
+
+def actionsDir = "${projectDir}/src/androidTest/java/sh/calaba/instrumentationbackend/actions"
+def assetsDir = "${buildDir}/assets"
+def versionFilePath = "${actionsDir}/version/Version.java"
+def actionsFile = "${assetsDir}/actions"
+def exampleManifestFile = "${projectDir}/../AndroidManifest.xml"
+def calabashJs = "${projectDir}/../calabash-js/src/calabash.js"
+def outputPath = "${buildDir}/outputs/apk/androidTest/debug/"
+def apkName = "app-debug-androidTest.apk"
+def serverApkName = "TestServer.apk"
+
+android {
+    compileSdkVersion 24
+    defaultConfig {
+        applicationId "sh.calaba.instrumentationbackend"
+        //noinspection MinSdkTooLow
+        minSdkVersion 8
+        //noinspection OldTargetApi
+        targetSdkVersion 24
+        versionCode 4
+        versionName "0.4.0"
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+    }
+    buildTypes {
+        release {
+            minifyEnabled false
+        }
+    }
+
+    sourceSets {
+        androidTestDebug {
+            assets.srcDirs += (new File(actionsFile)).getParentFile()
+        }
+    }
+}
+
+task notifyBuildStarted {
+    doLast {
+        println("Build started...")
+    }
+}
+
+task replaceVersion {
```

Comment:
I'd love to get get rid of this replacing in src files step. So how about getting rid of the ../../version file and use $(versionName). Write the value of $(versionName) to a resource file and read rewrite the version action to return the value read from the file

Reply:
I will keep original version file cause we are using this to create git tags as well. In addition, version in gradle is not aligned with common version.
Old version file will be copied into assets and we can read version in runtime.